### ISSUE TITLE
Import images

### ISSUE THREAD

**Issue by bferguso** [bcgov/BCHeritage#215]
Need to import the site images. The metadata has already been staged in #204 as bchp_staging.site_images but we are still waiting on access to the media itself to import it into the system.

For server sizing - there are approximately 6300 files totalling approximately 1.6Gb of storage.

Currently have loaded approximately 4350 files.

Deal with the following issues:
- [x] #348
- [x] #347
- [x] approximately 823 images that don't have entries in HRIA (eg DcRu-490) - Set title to "Missing metadata". List provided to TM by email. New ticket to be created if action required
- [x] #344
- [x] multiple references to the same file (eg documentkey 34874) - BF to investigate. Separate ticket? Find count and borden numbers for these. List provided to TM by email. New ticket to be created if action required

Unsupported file types are:
- [x] .cpg, .dbf, .sbn, .sbx, .shp, .shx, .xml, .prj 5 instances each, 5 sites, extra .prj file in 1 site - Move this to a separate ticket. Do we support .zip files? Provide Borden numbers for these in other ticket - TM - Don't need to store, but need to import as feature
- [x] .tif - 9 instances. Should we convert or see if we can support .tif filetypes? - Convert to jpg
- [x] .png - 7 instances. ~Should we convert or see if we can support.png filetype? - Convert to jpg~ These appear to be supported
- [x] #354

Existing image metadata type values are:
- Detailed Data Table
- Image-Drawing
- Image-Exterior
- Image-Historic
- Image-Interior
- Image Other
- Image(s), Other
- LINK
- Map
- Map, Detailed
- URL

proposing to clean up to the following list (Agreed to by TM):
- Detailed Data Table
- Image, Drawing
- Image, Exterior
- Image, Historic
- Image, Interior
- Image, Other
- Map
- Map, Detailed

Converted URL and LINK to Image, Other.

- [x] Confirm the above substitution is OK
- [x] Confirm that images uploaded on the HRIA site page were included in the import
- [x] #390
- [x] #395

**Comment by bferguso** [comment]
We now have access (almost 3 weeks later), however they have given me read/write not readonly. Waiting to see if we can get readonly access as originally requested. I've written a script to pull data off the drive but I'd rather not execute it until we've sorted out the access.

**Comment by TMcFarland-Heritage** [comment]
Image size: @KatieMcEvoy will look into our standard size for PHAD for us to consider if this will work for ARCHES as well. We want to be mindful of the cost of large images. We will also still need to ensure we can export to CRHP at their restricted file size.

Recommend we discuss the other outstanding issues at our next meeting: 
Deal with the following issues:

- mis-categorized files (eg EaQu-75 Pre-2021 Site Record.pdf is actually for a different non-BCRHP site)
-  unsupported image types (tif) **(Okay to convert these to JPG**)
-  approximately 823 images that don't have entries in HRIA (eg DcRu-490) - Set title to "Missing metadata" (**Very odd that it is such a high number, a bit disconcerting, is there any trend amongst the 823 images that don't have an entry in HRIA? DcRu-490 is the McPherson Theatre and does seem to have a record-let's discuss this one)**
-  multiple references to the same file (eg documentkey 34874) - BF to investigate. Separate ticket? Find count and borden numbers for these.

Agree to the following: proposing to clean up to the following list:

Detailed Data Table
Image, Drawing
Image, Exterior
Image, Historic
Image, Interior
Image, Other
Map
Map, Detailed
Converted URL and LINK to Image, Other. **(Question: is there another place in ARCHES for saving helpful links/URLs for historic places? e.g. link to the society's website who runs the site)**

**Comment by KatieMcEvoy** [comment]
The standard image size for PHAD is 2MB

**Comment by bferguso** [comment]
FYI - Storage costs should be much lower if we use S3 or another service. From my understanding there are a bunch of higher resolution images on the LAN somewhere - I think the goal should be to push those into the system and get rid of them from the LAN. Object storage will be much cheaper than LAN storage, and it probably makes sense to have all the images in one place if possible?

**Comment by bferguso** [comment]
@TMcFarland-Heritage, @KatieMcEvoy - Is there somewhere on the Site tab in HRIA that images can be uploaded, as well as the images form? I'm seeing the ability in the database to be able to link documents in the site form and wondering how that's used.

**Comment by KatieMcEvoy** [comment]
@bferguso when you right click the site Borden Number and follow the New ->Images pathway an image tab will be created under the site's Borden Number with fields for entering information and linking an image from the drive to HRIA.

**Comment by KatieMcEvoy** [comment]
I don’t think so but I’m not totally clear on it.

Katie
